sql/_05_plcsql/_01_testspec/_02_declaration/_01_parameter/_03_type_conversion/cases/02_01_03-14_sql_to_param_VARCHAR.sql

```diff
@@ -0,0 +1,210 @@
+
+call print_message('t_VARCHAR_VARCHAR. This scenario is a success.');
+create or replace procedure t_VARCHAR_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_VARCHAR_VARCHAR('VARCHAR', 'VARCHAR', 'VARCHAR CUBRID' ) ;
+drop procedure t_VARCHAR_VARCHAR ;
+--BUG
+
+call print_message('t_SET_VARCHAR. This scenario is a failure.');
+create or replace procedure t_SET_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_SET_VARCHAR('SET', 'VARCHAR', {'c','c','c','b','b','a'} ) ;
+drop procedure t_SET_VARCHAR ;
+--BUG
+
+call print_message('t_MULTISET_VARCHAR. This scenario is a failure.');
+create or replace procedure t_MULTISET_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_MULTISET_VARCHAR('MULTISET', 'VARCHAR', {'c','c','c','b','b','a'} ) ;
+drop procedure t_MULTISET_VARCHAR ;
+--BUG
+
+call print_message('t_LIST_VARCHAR. This scenario is a failure.');
+create or replace procedure t_LIST_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_LIST_VARCHAR('LIST', 'VARCHAR', {'c','c','c','b','b', 'a'} ) ;
+drop procedure t_LIST_VARCHAR ;
+--BUG
+
+call print_message('t_ENUM_VARCHAR. This scenario is a failure.');
+create or replace procedure t_ENUM_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_ENUM_VARCHAR('ENUM', 'VARCHAR', 'yellow' ) ;
+drop procedure t_ENUM_VARCHAR ;
+
+
+call print_message('t_BLOB_VARCHAR. This scenario is a failure.');
+create or replace procedure t_BLOB_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_BLOB_VARCHAR('BLOB', 'VARCHAR', BIT_TO_BLOB(X'000010') ) ;
+drop procedure t_BLOB_VARCHAR ;
+
+
+call print_message('t_CLOB_VARCHAR. This scenario is a failure.');
+create or replace procedure t_CLOB_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_CLOB_VARCHAR('CLOB', 'VARCHAR', CHAR_TO_CLOB('This is a Dog') ) ;
+drop procedure t_CLOB_VARCHAR ;
+
+
+call print_message('t_JSON_VARCHAR. This scenario is a failure.');
```

Comment:
It's a failure, but the answer file it was successful. Need to check.

Reply:
```suggestion
call print_message('t_JSON_VARCHAR. This scenario is a success.');
```
It is a success as in JSON to CHAR case.

Reply:
확인버젼 : CUBRID 11.4 (11.4.0.1339-962675d)

call t_JSON_CHAR('JSON', 'VARCHAR', json'{"a":1}' ) ;
위와 같이 호출 시에 아래와 같은 에러메세지가 출력됩니다.
ERROR: Stored procedure execute error: unsupported argument type - JSON

PLCSQL에서는 json타입은 미지원함으로 해당 케이스는 정상 처리하도록 하겠으며,
테스트 comment는 "failure"로 작성하겠습니다.